### PR TITLE
test: switch to smock package for partner proxy factory

### PR DESCRIPTION
**Pull request by LordCheta**

### CODE REVIEW

File: test/PartnerProxyFactory.spec.ts [hunk 1]
```diff
@@ -208,29 +199,29 @@ describe('Deploy PartnerProxyFactory, Create New Proxy Instances, Use new Partne
...
     );
 
-    await RIF.mock.transferFrom.returns(true);
+    RIF.transferFrom.returns(true);
 
-    await RIF.mock.approve.returns(true);
+    RIF.approve.returns(true);
 
-    await NodeOwner.mock.expirationTime.returns(EXPIRATION_TIME);
+    await NodeOwner.expirationTime.returns(EXPIRATION_TIME);
```
Comment (OnyekaIjeh) [inline test/PartnerProxyFactory.spec.ts:220]: `await` not required

File: test/PartnerProxyFactory.spec.ts [hunk 2]
```diff
@@ -41,58 +36,54 @@ async function initialSetup() {
...
-      rootNode: tldNode,
-    });
-
-  const { contract: PartnerProxy } = await deployContract<$PartnerProxy>(
-    '$PartnerProxy',
-    {}
+  const { contract: PartnerRegistrar } = await deployContract(
+    'PartnerRegistrar',
+    {
+      _nodeOwner: NodeOwner.address,
```
Comment (OnyekaIjeh) [inline test/PartnerProxyFactory.spec.ts:63]: There's inconsistency with these field names with the others

**Review by OnyekaIjeh — COMMENTED** [review]
Left some comments.
We should also consider removing `hardhat-exposed` and save us all these confusing `$` signs and creating a `deployContract` from smock for testing.

**Comment by sergio-jara** [comment]
> Left some comments. We should also consider removing `hardhat-exposed` and save us all these confusing `$` signs and replace them with a `deployContract` function using smock `mock` method for testing contracts.

**Comment by sergio-jara** [comment]
> Left some comments. We should also consider removing `hardhat-exposed` and save us all these confusing `$` signs and replace them with a `deployContract` function using smock `mock` method for testing contracts.

yes plis

File: test/PartnerProxyFactory.spec.ts (hunk 2 above)
Comment (LordCheta) [reply test/PartnerProxyFactory.spec.ts:63]: Missed this one, thanks for catching it

File: test/PartnerProxyFactory.spec.ts (hunk 1 above)
Comment (LordCheta) [reply test/PartnerProxyFactory.spec.ts:220]: Missed this one, thanks for catching it

**Review by sergio-jara — APPROVED** [review]

**Pull request merged**